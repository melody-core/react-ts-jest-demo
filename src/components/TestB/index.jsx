/*
 * @Author: 六弦(melodyWxy)
 * @Date: 2022-02-14 15:26:07
 * @LastEditors: 六弦(melodyWxy)
 * @LastEditTime: 2022-02-14 15:28:11
 * @FilePath: /node-test-js-demo/Users/wxy/codeWorks/git-demos/git-test/react-study/src/components/TestB/index.jsx
 * @Description: update here
 */

import { Component } from "react";
class TestB extends Component {
  state = {
    count: 0,
  };

  handleClick = () => {
    this.setState({
      count: this.state.count + 1,
    });
  };

  render() {
    const { count } = this.state;
    return (
      <div className="count-wrap" onClick={this.handleClick}>
        {count}
      </div>
    );
  }
}

export default TestB;
