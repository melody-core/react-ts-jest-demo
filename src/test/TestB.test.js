/*
 * @Author: 六弦(melodyWxy)
 * @Date: 2022-02-14 15:37:56
 * @LastEditors: 六弦(melodyWxy)
 * @LastEditTime: 2022-02-15 09:14:25
 * @FilePath: /debugger/Users/wxy/codeWorks/git-demos/git-test/react-study/src/test/TestB.test.js
 * @Description: update here
 */

import TestB from "../components/TestB";
import { render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';


beforeEach(()=>{
    console.log('---------beforeEach-------')
})

beforeAll(()=>{
    console.log('----beforeAll-----')
})
test("TestB组件渲染功能正常", ()=> {
    render(<TestB />);
    const target = document.body.querySelector(".count-wrap");
    expect(target).not.toBeNull();
})

test("TestB组件交互功能正常", ()=> {
    render(<TestB />);
    const target = document.body.querySelector(".count-wrap");
    expect( target.innerHTML ).toEqual("0");
    fireEvent.click(target);
    fireEvent.click(target);
    expect( target.innerHTML ).toEqual("2");
})

test("TestB组件内置回调功能正常", ()=> {
    const component = renderer.create(
        <TestB />
    )
    const instance = component.getInstance();
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(instance.state.count).toEqual(0);
    instance.handleClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(instance.state.count).toEqual(1);
})
