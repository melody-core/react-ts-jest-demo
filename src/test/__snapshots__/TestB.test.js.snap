// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestB组件内置回调功能正常 1`] = `
<div
  className="count-wrap"
  onClick={[Function]}
>
  0
</div>
`;

exports[`TestB组件内置回调功能正常 2`] = `
<div
  className="count-wrap"
  onClick={[Function]}
>
  1
</div>
`;
